src/components/pokemon/pokemon.model.ts

```diff
@@ -49,6 +49,56 @@ class Pokemon implements Identifyable {
      */
     public buddySize: Identifyable;
     public modelHeight: number;
+
+    /**
+     * What could the player choose to evolve this pokemon into?
+     */
+    public nextEvolutionBranches: Identifyable[];
+
+    /**
+     * All the possible future evolutions of this pokemon, including indirect
+     * evolutions. For example, given Charmander, futureEvolutions would list
+     * Charmeleon and Charizard.
+     */
+    public futureEvolutions: EvolutionTree;
+
+    /**
+     * All past evolutions of this pokemon, direct and indirect.
+     *
+     * For Charizard, this field would list Charmander and Charmeleon.
+     */
+    public pastEvolutions: Identifyable[];
+}
+
+class EvolutionTree implements Identifyable {
+    /**
+     * An EvolutionTree represents all possible future evolutions of a pokemon.
+     * Mutually exclusive choices are given on the same level, while sequential
+     * choices are presented in increasing depth.
+     *
+     * For example, Eevee's EvolutionTree would take 6 objects in depth 2, but
+     * Charmander's EvolutionTree would take 3 objects in depth 3.
+     */
+    public name: string;
+    public id: string;
+    public futureEvolutions: EvolutionTree[] = [];
```

Comment:
Your approach, as described, won't handle [Wurmple](http://bulbapedia.bulbagarden.net/wiki/Wurmple_(Pok%C3%A9mon)#Evolution). However, that's not what I'm actually worried about; a dictionary would take care of Wurmple.

What I'm actually worried about is a baby pokemon for Eevee or Wurmple, or a similar delayed-branching structure. Something like P1 -> P2 -> (P3 OR P4).

This was a hard problem, and I didn't want to use a recursive tree, let alone create my own hacky tree data type. It's simply the most direct solution that solves any case that could possibly happen.

If the integer `evolutionStage` can be replaced with a delimited string or an array of IDs representing the path through the tree needed, then maybe a flat list could work. Allowing duplicate `evolutionStages` is not enough to handle Wurmple fully (ie, be able to tell that Silcoon evolves into Beautifly vs. Cascoon evolves into Dustox).

(And yes, I know Wurmple isn't in Pokemon Go yet. They have Fairy type in Pokemon Go, so I am going to assume it's a matter of *when* they get to it, not *if* they'll get to it. I'd rather not set myself up for maintenance work when I know reasonably well that I'm going to need a more general solution.)